### PR TITLE
Add changelog_from_git_commits action

### PR DESCRIPTION
**Pull request by mfurtak**
@KrauseFx Can we get your thoughts on this?

### CODE REVIEW

File: lib/fastlane/actions/changelog_from_git_commits.rb [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+      def self.available_options
+        [
+          FastlaneCore::ConfigItem.new(key: :between,
+                                       env_name: 'FL_GIT_COLLECT_MESSAGES_BETWEEN',
+                                       description: 'Array containing two Git revision values between which to collect messages',
+                                       optional: true,
+                                       is_string: false,
+                                       verify_block: proc do |value|
+                                         raise "'between' must be an array of size 2".red unless value and value.size == 2
```
Comment (KrauseFx) [inline lib/fastlane/actions/changelog_from_git_commits.rb]: It is recommended to not use the `and` keyword after an inline `if` or `unless`
```ruby
unless value and value.size == 2
```
instead it's safer to use 
```ruby
unless value && value.size == 2
```
There is a different in the priority of those 2 keywords.

To make it more Ruby like you could even use

```ruby
It is recommended to not use the `and` keyword after an inline `if` or `unless`
```ruby
unless value and value.size == 2
```
instead it's safer to use 
```ruby
unless value && value.size == 2
```ruby
unless (value || []).count == 2
```
also you could add a check for the type before that line to fail if the user passes e.g. a number

```ruby
raise ":between must be of type array" unless value.kind_of?(Array)
```

:+1:

File: lib/fastlane/actions/changelog_from_git_commits.rb [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+                                       is_string: false)
+        ]
+      end
+
+      def self.return_value
+        "Returns a String containing your formatted git commits"
+      end
+
+      def self.author
+        'fastlane team'
```
Comment (KrauseFx) [inline lib/fastlane/actions/changelog_from_git_commits.rb]: You can add your names to this, define multiple authors using 
```ruby
["user1", "user2"]
```
this makes actions more personal and also makes it easier to find the right person if someone needs help with an action :+1:

Comment (KrauseFx) [inline lib/fastlane/actions/changelog_from_git_commits.rb]: The author is shown when running `fastlane actions` or `fastlane action changelog_from_git_commits`

File: lib/fastlane/helper/git_helper.rb [hunk 3]
```diff
@@ -1,17 +1,52 @@
...
+    def self.git_author_email
+      s = last_git_commit_formatted_with('%ae')
       return s if s.to_s.length > 0
       return nil
     rescue
       return nil
     end
 
+    # Returns the unwrapped subject and body of the last commit
+    # <b>DEPRECATED:</b> Use <tt>last_git_commit_message</t> instead.
```
Comment (KrauseFx) [inline lib/fastlane/helper/git_helper.rb]: Nice :+1:

File: lib/fastlane/actions/changelog_from_git_commits.rb (hunk 1 above)
Comment (mfurtak) [inline lib/fastlane/actions/changelog_from_git_commits.rb]: Great, will do.

File: lib/fastlane/actions/changelog_from_git_commits.rb (hunk 2 above)
Comment (mfurtak) [inline lib/fastlane/actions/changelog_from_git_commits.rb]: I love it!

**Comment by KrauseFx** [comment]
:rocket:

**Pull request merged**